From c1b5549c8a902daf512e8f3ce2e40e9225f2a65c Mon Sep 17 00:00:00 2001
From: Oleksandr Tyshchenko <oleksandr_tyshchenko@epam.com>
Date: Wed, 29 Aug 2018 17:53:48 +0300
Subject: [PATCH] HACK: soc: Extend always on logic to H3 ES3.0

In order to get systems fucntional under XEN hypevisor owned all
IPMMUs, we need linux to not disable any power domains for all SOCs.

Signed-off-by: Andrii Anisov <andrii_anisov@epam.com>
Signed-off-by: Oleksandr Tyshchenko <oleksandr_tyshchenko@epam.com>
---
 drivers/soc/renesas/rcar-sysc.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/soc/renesas/rcar-sysc.c b/drivers/soc/renesas/rcar-sysc.c
index a063887..dc2b974 100644
--- a/drivers/soc/renesas/rcar-sysc.c
+++ b/drivers/soc/renesas/rcar-sysc.c
@@ -66,6 +66,12 @@
 static
 const struct soc_device_attribute rcar_sysc_quirks_match[] __initconst = {
 	{
+		.soc_id = "r8a7795", .revision = "ES3.0",
+		.data = (void *)(BIT(R8A7795_PD_A3VP) | BIT(R8A7795_PD_CR7)
+			| BIT(R8A7795_PD_A3VC) | BIT(R8A7795_PD_A2VC0)
+			| BIT(R8A7795_PD_A2VC1) | BIT(R8A7795_PD_A3IR)),
+	},
+	{
 		.soc_id = "r8a7795", .revision = "ES2.0",
 		.data = (void *)(BIT(R8A7795_PD_A3VP) | BIT(R8A7795_PD_CR7)
 			| BIT(R8A7795_PD_A3VC) | BIT(R8A7795_PD_A2VC0)
-- 
2.7.4

